commit d93ebd7f85d1010f97bc4dc182677b04ee9bace1
Author: Samuel Thibault <samuel.thibault@ens-lyon.org>
Date:   Tue Aug 25 21:20:52 2026 +0200

    hurd: Warn about setting _TIME_BITS=64 on i386 GNU Hurd
    
    We do not plan to implement compatibility layers etc.
    
    Other 32b Hurd ports will always use 64b time_t.

diff --git a/sysdeps/mach/hurd/i386/features-time64.h b/sysdeps/mach/hurd/i386/features-time64.h
new file mode 100644
index 0000000000..5da6696e50
--- /dev/null
+++ b/sysdeps/mach/hurd/i386/features-time64.h
@@ -0,0 +1,23 @@
+/* Features part to handle 64-bit time_t support.  Hurd/i386 version.
+   Copyright (C) 2021-2026 Free Software Foundation, Inc.
+   This file is part of the GNU C Library.
+
+   The GNU C Library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Lesser General Public
+   License as published by the Free Software Foundation; either
+   version 2.1 of the License, or (at your option) any later version.
+
+   The GNU C Library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Lesser General Public License for more details.
+
+   You should have received a copy of the GNU Lesser General Public
+   License along with the GNU C Library; if not, see
+   <https://www.gnu.org/licenses/>.  */
+
+#if defined _TIME_BITS
+# if _TIME_BITS == 64
+#  warning "_TIME_BITS=64 is not supported on i386 GNU Hurd"
+# endif
+#endif
